#include "addlessonnameform.h"
#include "ui_addlessonnameform.h"
#include"FileManagerCore/managerfs.h"
#include<QShortcut>

AddLessonNameForm::AddLessonNameForm(QWidget *parent)
    : QWidget(parent)
    , ui(new Ui::AddLessonNameForm)
    , schedule(std::make_unique<DataSchedule>())
    , parseJson(std::make_unique<ScheduleJsonParser>())
{
    ui->setupUi(this);
    this->setLayout(ui->gridLayout);

    ManagerWorker w;
    w.CreateDir(".","–†–∞—Å–ø–∏—Å–∞–Ω–∏–µ");

    QShortcut* shortcut = new QShortcut(QKeySequence("Enter"),this);

    connect(ui->btnSave,&QPushButton::clicked,this,[this](){

        if(!ui->lineEditNamePair->text().isEmpty()){
            ui->listWidget->addItem("–ù–∞–∑–≤–∞–Ω–∏–µ:" +ui->lineEditNamePair->text());
            ui->lineEditNamePair->setText("");
        }

    });
    connect(shortcut,&QShortcut::activated,this,[this](){

        if(!ui->lineEditNamePair->text().isEmpty()){
            ui->listWidget->addItem("–ù–∞–∑–≤–∞–Ω–∏–µ:" + ui->lineEditNamePair->text());
            ui->lineEditNamePair->setText("");
        }

    });


    connect(ui->btnGo,&QPushButton::clicked,this,&AddLessonNameForm::SendLessonNameDataToJson);

    connect(ui->btnHelp,&QPushButton::clicked,this,[this](){

            QMessageBox::information(this,"üìö ARDA - –ü—Ä–∞–≤–∏–ª–∞ —Å–∏—Å—Ç–µ–º—ã",
                "<html>"
                "<head>"
                "<style>"
                "body { font-family: 'Segoe UI', Arial; font-size: 10pt; }"
                ".header { color: #2C3E50; font-size: 12pt; font-weight: bold; text-align: center; margin-bottom: 15px; }"
                ".rule { background: #F8F9FA; padding: 10px; border-radius: 6px; margin: 8px 0; border-left: 4px solid; }"
                ".rule1 { border-color: #E74C3C; }"
                ".rule2 { border-color: #3498DB; }"
                ".title { color: #34495E; font-weight: bold; margin-bottom: 5px; }"
                ".text { color: #7F8C8D; }"
                ".tip { background: #FFFDE7; padding: 8px; border-radius: 4px; margin-top: 10px; border: 1px solid #FFEB3B; font-size: 9pt; color: #F39C12; }"
                "</style>"
                "</head>"
                "<body>"
                "<div class='header'>üéì –û—Å–Ω–æ–≤–Ω—ã–µ –ø—Ä–∞–≤–∏–ª–∞ —Ä–∞–±–æ—Ç—ã —Å ARDA</div>"

                "<div class='rule rule1'>"
                "<div class='title'>1Ô∏è‚É£ –£–ù–ò–ö–ê–õ–¨–ù–û–°–¢–¨ –ù–ê–ó–í–ê–ù–ò–ô</div>"
                "<div class='text'>‚Ä¢ –í—Å–µ –Ω–∞–∑–≤–∞–Ω–∏—è –ø–∞—Ä –¥–æ–ª–∂–Ω—ã –±—ã—Ç—å <b>—É–Ω–∏–∫–∞–ª—å–Ω—ã–º–∏</b><br>"
                "‚Ä¢ –ü—Ä–∏ –¥–æ–±–∞–≤–ª–µ–Ω–∏–∏ –¥—É–±–ª–∏–∫–∞—Ç–∞ —Å–∏—Å—Ç–µ–º–∞ <b style='color:#C0392B;'>–∞–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∏ —É–¥–∞–ª—è–µ—Ç</b> —Å—Ç–∞—Ä—É—é –∑–∞–ø–∏—Å—å<br>"
                "‚Ä¢ –ù–æ–≤–∞—è –ø–∞—Ä–∞ —Å–æ—Ö—Ä–∞–Ω—è–µ—Ç—Å—è —Å —É–Ω–∏–∫–∞–ª—å–Ω—ã–º –∏–¥–µ–Ω—Ç–∏—Ñ–∏–∫–∞—Ç–æ—Ä–æ–º</div>"
                "</div>"

                "<div class='rule rule2'>"
                "<div class='title'>2Ô∏è‚É£ –£–î–ê–õ–ï–ù–ò–ï –ü–ê–†</div>"
                "<div class='text'>‚Ä¢ –î–ª—è —É–¥–∞–ª–µ–Ω–∏—è <b>–∫–ª–∏–∫–Ω–∏—Ç–µ –ø–æ –Ω–∞–∑–≤–∞–Ω–∏—é</b> –ø–∞—Ä—ã –≤ —Å–ø–∏—Å–∫–µ<br>"
                "‚Ä¢ –°–∏—Å—Ç–µ–º–∞ –∑–∞–ø—Ä–æ—Å–∏—Ç –ø–æ–¥—Ç–≤–µ—Ä–∂–¥–µ–Ω–∏–µ –æ–ø–µ—Ä–∞—Ü–∏–∏<br>"
                "‚Ä¢ –£–¥–∞–ª–µ–Ω–Ω—ã–µ –¥–∞–Ω–Ω—ã–µ –º–æ–∂–Ω–æ –≤–æ—Å—Å—Ç–∞–Ω–æ–≤–∏—Ç—å –∏–∑ –∞—Ä—Ö–∏–≤–∞</div>"
                "</div>"

                "<div class='tip'>"
                "üí° <i>–ü–æ–¥—Å–∫–∞–∑–∫–∞: –î–ª—è –±—ã—Å—Ç—Ä–æ–≥–æ –ø–æ–∏—Å–∫–∞ –∏—Å–ø–æ–ª—å–∑—É–π—Ç–µ –∫–æ–º–±–∏–Ω–∞—Ü–∏—é Ctrl+F</i>"
                "</div>"
                "</body>"
                "</html>"
                );

    });
}

AddLessonNameForm::~AddLessonNameForm()
{
    delete ui;
}

bool AddLessonNameForm::SendLessonNameDataToJson()
{
    if(ui->listWidget->count() == 0){
        QMessageBox::warning(this,"–û—à–∏–±–∫–∞ –æ—Ç–ø—Ä–∞–≤–∫–∏","–ê –ø–∞—Ä—ã –≥–¥–µ?");
        return false;

    }

    for(int i = 0; i<ui->listWidget->count(); i++)
    {
        QString str = std::move(ui->listWidget->item(i)->text().remove("–ù–∞–∑–≤–∞–Ω–∏–µ:"));
        schedule->LessonsName.insert(str);
    }

    QString file = "–†–∞—Å–ø–∏—Å–∞–Ω–∏–µ/lesson_name.json";

    bool success = parseJson->setLessonNameCabinets(file,schedule->LessonsName);

    if(!success)
    {
        QMessageBox::warning(this,"–û—à–∏–±–∫–∞ –æ—Ç–ø—Ä–∞–≤–∫–∏","—á—Ç–æ —Ç–æ –≤—Å—Ç–∞–≤–∏—Ç—å –¥–∞–Ω–Ω—ã–µ –Ω–µ –ø–æ–ª—É—á–∏—Å–ª–æ—Å—å\n\n –ü—Ä–æ–≤–µ—Ä—å –ø–∞–ø–∫—É –†–∞—Å–ø–∏—Å–∞–Ω–∏–µ –ø–æ –ø—É—Ç–∏ –ø—Ä–æ–≥—Ä–∞–º–º—ã –∏–ª–∏ –ø–µ—Ä–µ–∑–∞–ø—É—Å—Ç–∏ –ø—Ä–æ–≥—É");

        return success;
    }

    QMessageBox::information(this,"–£—Å–ø–µ—à–Ω–æ","–î–æ–±–∞–≤–ª–µ–Ω–æ —É—Å–ø–µ—à–Ω–æ");

    return success;


}
